content/Software Support/IDE Settings/Select-programmer-in-Arduino-IDE.md

```diff
@@ -0,0 +1,17 @@
+---
+title: "Select programmer in Arduino IDE"
+---
+
+When you have [selected a board in Arduino IDE](https://support.arduino.cc/hc/en-us/articles/4406856349970-Select-board-and-port-in-Arduino-IDE#board), external programmers can be selected under _Tools > Programmer_ in the menu bar. The options are defined in the platform's `programmers.txt` file[^1].
+
+[^1]: [Platform specification – programmers.txt](https://arduino.github.io/arduino-cli/latest/platform-specification#programmerstxt)
+
+The programmer selection is used by:
+
+* The _Tools > Burn Bootloader_ menu bar item.
+* The _Sketch > Upload Using Programmer_ option.
+
+The programmer selection does **not** affect:
+
+* Sketch compilation (for example, when clicking the ![Verify button](img/symbol_verify2.png) **Verify** button).
+* Uploading a sketch to a board connected to your computer by clicking the the ![Upload button](img/symbol_upload2.png) **Upload** button.
```

Comment:
This is incorrect for some boards. It is possible to configure the board definition so that the "**Upload**" button acts as **Sketch > Upload Using Programmer**:

https://arduino.github.io/arduino-cli/0.27/platform-specification/#upload-using-programmer-by-default

An example of such a board definition is ["**Arduino Gemma**"](https://github.com/arduino/ArduinoCore-avr/blob/1.8.5/boards.txt#L1069-L1095)

If you are experimenting with this feature, note that support for it has not yet been added to Arduino IDE 2.x:

https://github.com/arduino/arduino-ide/issues/103

Reply:
@per1234 How's this?

https://github.com/arduino/help-center-content/pull/120/commits/18742d99200637b4266fd92bd7d71768bb8c4c7b